RELEASE_PROCESS.md

```diff
@@ -0,0 +1,51 @@
+# Release Process
+
+This document contains all the necessary information about out release process.
+
+## Release branch
+
+For each release we create a separate release branch.
```

Comment:
I've thought about that, but I was discouraged because all contributors would have to figure out which branch to PR against. It'd add an extra communication complexity where we have to tell them something like:

"Hey could you reopen this PR against the 1.17.1 branch, we'd like to include this PR in the upcoming release"

This is a bigger issue with patches that needs to be released asap without communication blocks